### ISSUE TITLE
How to apply a legacy config to specific files

### ISSUE THREAD

**Issue by bmish** [eslint/eslintrc#140]
For example, applying a test config like `mocha/recommended` or `jest/recommended` only to test files seems like a fairly common practice. But I haven't seen any official documentation on this.

I found one way to do it here:

```js
module.exports = [
    // Apply mocha config only to tests.
    ...compat.extends("plugin:mocha/recommended").map(config => ({ ...config, files: ["tests/**/*.js"] })),
];
```

Another similar example:

https://github.com/psychobolt/vite-storybook-boilerplate/blob/1356effb9bdd6f8d09a8c5557e4a8aa411904c68/packages/commons/eslint.config.ts#L20-L23

It may be worth documenting an example like this in the README.

**Comment by privatenumber** [comment]
Another solution may to be to iterate over the configs and extend the previous config for each config, in case there are more properties than `files` that needs to be inherited:

```ts
const configs = flatCompat.extends('plugin:vue/vue3-recommended')

// extend previous config
configs.forEach((config, index) => {
	const prev = configs[index - 1]
	if (prev) {
		Object.assign(config, { ...prev, ...config })
	}
})
```

I noticed in https://github.com/vuejs/eslint-plugin-vue, one of the base configs sets the `processor` and that also needs to be inherited by the configs. Otherwise, the configs without `files` and `processor` (last ones) will try to apply vue rules to non-Vue files emit a parsing error:

```ts
[
  {
    files: [ '**/*.vue' ],
    processor: {
      preprocess: [Function: preprocess],
      postprocess: [Function: postprocess],
      supportsAutofix: true,
      meta: [Object]
    }
  },
  ...,
  {
    rules: {
      'vue/attributes-order': 'warn',
      'vue/component-tags-order': 'warn',
      'vue/no-lone-template': 'warn',
      'vue/no-multiple-slot-args': 'warn',
      'vue/no-v-html': 'warn',
      'vue/order-in-components': 'warn',
      'vue/this-in-template': 'warn'
    }
  }
]
```

**Comment by nzakas** [comment]
I actually think such documentation should be in the main docs, because it applies to flat config arrays and not uniquely to legacy configs.

**Comment by bmish** [comment]
Good point, I'm open to generalizing this to any config array and transferring the issue to the main ESLint repo.
